### ISSUE TITLE
List recent code contributors in footer, using "name-your-contributors" script

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#931]
This has been marked as a good candidate for becoming a [first-timers-only issue](http://www.firsttimersonly.com) like [these](https://github.com/publiclab/plots2/labels/first-timers-only), meaning that it's simple, self-contained, and with some extra formatting, could be a great entry point for a new contributor. If you're familiar enough with this code, please consider [reformatting or reposting it as a first-timers-only issue](https://publiclab.org/notes/warren/10-31-2016/create-a-welcoming-first-timers-only-issue-to-invite-new-software-contributors), and then ping @jywarren to get it labelled.

****

We're (@david-days and I) were thinking that, as an outreach technique, we could run https://github.com/richardlitt/name-your-contributors and actually list recent contributors (including issue submitters and commenters) in the footer, to try to highlight their work, and to attract new code contributors.

We should make a sub-template called `_footer_contributors.html.erb` and include it in the appropriate section of the footer with a message like:

> This site is created and maintained by community contributors, from designers to coders. Learn more and [get involved here](). Contributors over the past month include [(a)person](), [(a)person](), [(a)person](), [(a)person](), [(a)person]().

Is there a way also to specifically invite, using [Outreachy's](https://www.gnome.org/outreachy/) language, "women (cis and trans), trans men, and genderqueer people, Black/African American, Hispanic/Latin@, American Indian, Alaska Native, Native Hawaiian, and Pacific Islander people"? Perhaps that needs to be on its own page, as it's a longer string of text, but the Outreachy program has found that specific invitation in those words makes a big difference to inviting a diverse contributor group. 

Perhaps we could use the shorter text from their front page in the footer text:

>  people from groups underrepresented in free and open source software

****

## Update

This means adding this:

```ruby
<% cache('feature_footer-contributors') do %> // <== this caches and uses a cached version of the feature, which is expired when the feature is edited
  <%= feature('footer-contributors') %>
<% end %>
```

Just before this line: 

https://github.com/publiclab/plots2/blob/master/app/views/layouts/_footer.html.erb#L9

**Comment by jywarren** [comment]
Left a bug report when ran out of memory while running name-your-contributors: https://github.com/RichardLitt/name-your-contributors/issues/15

**Comment by jywarren** [comment]
I like this -- we should make a sub-template called `_footer_contributors.html.erb` and include it in the appropriate section of the footer with a message like:

> This site is created and maintained by community contributors, from designers to coders. Learn more and [get involved here](). Contributors over the past month include [(a)person](), [(a)person](), [(a)person](), [(a)person](), [(a)person]().

Is there a way also to specifically invite, using [Outreachy's](https://www.gnome.org/outreachy/) language, "women (cis and trans), trans men, and genderqueer people, Black/African American, Hispanic/Latin@, American Indian, Alaska Native, Native Hawaiian, and Pacific Islander people"? Perhaps that needs to be on its own page, as it's a longer string of text, but the Outreachy program has found that specific invitation in those words makes a big difference to inviting a diverse contributor group.

**Comment by jywarren** [comment]
Perhaps we could use the shorter text from their front page in the footer text:

>  people from groups underrepresented in free and open source software

**Comment by ebarry** [comment]
I respect Outreachy and could be convinced to take their word for it but i was surprised to see the demographic categories from the US Census being used as welcoming text.

**Comment by jywarren** [comment]
I don't think the US Census uses all of those terms, and I know Outreachy spent a good deal of time testing out this language with people. Do you have suggestions to change it, or would you like to contact the Outreachy team for more on their reasoning and process?

**Comment by ebarry** [comment]
I wonder if ConductCom would be interested in inquiring with Outreachy about this?

the terms Black/African American, American Indian, Alaska Native, Native Hawaiian, and Pacific Islander people seem like a federal United States viewpoint, these exact terms are from the 2010 US Census with the clarification that Hispanic/Latin@ is an ethnicity not a race category https://www.census.gov/quickfacts/meta/long_RHI425215.htm I would love to mention underrepresented groups, but we are global and this list seems very US focused.

**Comment by ebarry** [comment]
I am going to reach out to ConductCom about this.

**Comment by jywarren** [comment]
This is Outreachy's US program, yes!

**Comment by ryzokuken** [comment]
@jywarren Using `GITHUB_OGN_TOKEN=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX name-your-contributors publiclab --since=2016-01-01T00:00:00Z` gives the following output.

```
Got response
wrote issue_creators
wrote issue_commenters
wrote pr_creators
wrote pr_reviewers
wrote commenters
Done collecting data.
- [@ajessee](//github.com/ajessee) (André)
- [@alexey-klimuk](//github.com/alexey-klimuk) (Alexey)
- [@ananyo2012](//github.com/ananyo2012) (Ananya Maiti)
- [@annhchen](//github.com/annhchen) (Ann Chen)
- [@ap916](//github.com/ap916) (Abhishek Pandey)
- [@batiste](//github.com/batiste) (Batiste Bieler)
- [@blakeyoder](//github.com/blakeyoder) (Blake Yoder)
- [@btbonval](//github.com/btbonval) (Bryan Bonvallet)
- [@carolineh101](//github.com/carolineh101) (Caroline Ho)
- [@ccloquet](//github.com/ccloquet) (Christophe Cloquet)
- [@Charmatzis](//github.com/Charmatzis) (Christos)
- [@chongr](//github.com/chongr)
- [@cpdavngr](//github.com/cpdavngr)
- [@darshan934](//github.com/darshan934) (Darshan)
- [@david-days](//github.com/david-days) (David C Days)
- [@dbeavers](//github.com/dbeavers) (Dan Beavers)
- [@dcanprdn](//github.com/dcanprdn)
- [@dhaffnersr](//github.com/dhaffnersr)
- [@dhenrygithub](//github.com/dhenrygithub) (Dan Henry)
- [@dwblair](//github.com/dwblair)
- [@ebarry](//github.com/ebarry) (Liz Barry)
- [@ethanbass](//github.com/ethanbass)
- [@Fastie](//github.com/Fastie)
- [@fiestin](//github.com/fiestin)
- [@foggy1](//github.com/foggy1) (Austin Lanari)
- [@franktank](//github.com/franktank) (Franky Liang)
- [@gcttong](//github.com/gcttong) (Gary Tong)
- [@genericlady](//github.com/genericlady) (Yianna)
- [@Goldtean](//github.com/Goldtean) (Jeremy Oltean)
- [@gr2m](//github.com/gr2m) (Gregor Martynus)
- [@gretchengehrke](//github.com/gretchengehrke)
- [@IanThorp](//github.com/IanThorp) (Ian Thorp)
- [@icarito](//github.com/icarito) (Sebastian Silva)
- [@imasaka1117](//github.com/imasaka1117)
- [@Jefferson-Faseler](//github.com/Jefferson-Faseler) (Jefferson Faseler)
- [@jiteshjha](//github.com/jiteshjha) (Jitesh Kumar Jha)
- [@justinffs](//github.com/justinffs) (Justin Hamade)
- [@justinmanley](//github.com/justinmanley) (Justin Manley)
- [@jywarren](//github.com/jywarren) (Jeffrey Warren)
- [@lalithr95](//github.com/lalithr95) (Lalith Rallabhandi)
- [@manish310794](//github.com/manish310794) (Manish Agarwal)
- [@markfox1](//github.com/markfox1) (Mark)
- [@mathewlippincott](//github.com/mathewlippincott) (mathew lippincott)
- [@mnorton-ffs](//github.com/mnorton-ffs) (Michael Norton)
- [@mvayngrib](//github.com/mvayngrib) (Mark Vayngrib)
- [@mzishtiaq](//github.com/mzishtiaq) (Zeeshan Ishtiaq)
- [@neeleshn](//github.com/neeleshn) (Neelesh Nidadhavolu)
- [@neliojrr](//github.com/neliojrr) (Nélio Carneiro)
- [@nickstaggs](//github.com/nickstaggs) (Nick Staggs)
- [@numeroteca](//github.com/numeroteca) (numeroteca)
- [@patcoyle](//github.com/patcoyle)
- [@pdhixenbaugh](//github.com/pdhixenbaugh)
- [@phillip-hogan](//github.com/phillip-hogan) (Phillip Hogan)
- [@prabhagharan](//github.com/prabhagharan)
- [@rafael-acerqueira](//github.com/rafael-acerqueira) (Rafael Aquino)
- [@rmeister](//github.com/rmeister) (Richard Meister)
- [@robotp](//github.com/robotp)
- [@RodCardenas](//github.com/RodCardenas) (Rodrigo Cardenas)
- [@ruan-brandao](//github.com/ruan-brandao) (Ruan Brandão)
- [@ryzokuken](//github.com/ryzokuken) (Ujjwal Sharma)
- [@sandyvern](//github.com/sandyvern) (Sandy)
- [@sdosemagen](//github.com/sdosemagen) (Shannon Dosemagen)
- [@sethwoodworth](//github.com/sethwoodworth) (Seth Woodworth)
- [@shaief](//github.com/shaief) (Shai Efrati)
- [@shapironick](//github.com/shapironick)
- [@shizi](//github.com/shizi) (Nenad Natošević)
- [@sjones88](//github.com/sjones88) (Steven Jones)
- [@skilfullycurled](//github.com/skilfullycurled)
- [@sofiaclara93](//github.com/sofiaclara93) (Sofia Cepeda)
- [@Squawk09](//github.com/Squawk09)
- [@Sreyanth](//github.com/Sreyanth) (Sreyanth)
- [@steviepubliclab](//github.com/steviepubliclab)
- [@Stoft1](//github.com/Stoft1)
- [@sudheesh001](//github.com/sudheesh001) (Sudheesh Singanamalla)
- [@tonypubliclab](//github.com/tonypubliclab)
- [@triat](//github.com/triat) (Tom Riat)
- [@ujithaperera](//github.com/ujithaperera) (Ujitha Perera)
- [@viechdokter](//github.com/viechdokter)
- [@vzhz](//github.com/vzhz) (Veronica Hanus)
- [@WagnerLoc](//github.com/WagnerLoc) (Stephan Wagner)
- [@wasnotrice](//github.com/wasnotrice) (Eric Watson)
- [@wly731](//github.com/wly731) (Victor Wong)
- [@ykl7](//github.com/ykl7)
- [@YogiBGitHub](//github.com/YogiBGitHub)
- [@ZettStai](//github.com/ZettStai)
```

Considering my own name was included, I assume the tool worked as needed. 
I suppose all we need to do now is to parse this list and render a few random names from the list on the page?

**Comment by jywarren** [comment]
OMG!!! WOW

I should try running it with the same command... don't know why it didn't
work for me. THANKS!

**Comment by ryzokuken** [comment]
@jywarren I'm sure you'll take care of the Access Token. I did not reveal it for security reasons, I'm sure you'd understand.

**Comment by jywarren** [comment]
yes - i'd tried to save the token in my `.bash_profile` but maybe it didn't work. How long did it take you to run this?

**Comment by ryzokuken** [comment]
a lot of time, actually. Not much over the network, but a lot of it while computing data after it had downloaded. It did make me re-think if we're really going to use it in production. (Imagine this running everytime the server renders)

**Comment by jywarren** [comment]
I'm trying to run it now, just for October. The 85 yours returned is too
long anyways to put in the footer!

**Comment by ryzokuken** [comment]
I did have an idea though, we could cleverly transform this into "Contributors this month", running the command and saving the list once every month on the server and using that list to render the contributors for the following month, which would significantly make it faster (we're only fetching data we already saved)

**Comment by jywarren** [comment]
We could use a Feature, and not actually include the names in the codebase -- the same as in #932 -- that way we don't have to do it at the same time as committing new code.

**Comment by jywarren** [comment]
I think that with #932's code example, and a Feature named "recent-contribtuors", we could turn this into a pretty actionable `first-timers-only` issue, or someone who's recently joined the project could take it on!

**Comment by jywarren** [comment]
I went ahead and submitted a fix for this just now -- in #1151!
